### PR TITLE
Support Ecto.Query.API.json_extract_path/2

### PR DESCRIPTION
**Pull request by wojtekmach**
Ref https://github.com/elixir-ecto/ecto/pull/3227

### CODE REVIEW

File: lib/ecto/adapters/myxql/connection.ex [hunk 1]
```diff
@@ -551,6 +551,19 @@ if Code.ensure_loaded?(MyXQL) do
 
     defp expr({:count, _, []}, _sources, _query), do: "count(*)"
 
+    defp expr({:json_extract_path, _, [expr, path]}, sources, query) do
+      path =
+        Enum.map(path, fn
+          binary when is_binary(binary) ->
+            [".", binary]
```
Comment (michalmuskala) [inline lib/ecto/adapters/myxql/connection.ex]: Do we need to do some sort of escaping here?

File: lib/ecto/adapters/postgres/connection.ex [hunk 2]
```diff
@@ -569,6 +569,15 @@ if Code.ensure_loaded?(Postgrex) do
        interval(count, interval, sources, query) | ")::date"]
     end
 
+    defp expr({:json_extract_path, _, [expr, path]}, sources, query) do
+      path =
+        path
+        |> Enum.map(&to_string/1)
```
Comment (michalmuskala) [inline lib/ecto/adapters/postgres/connection.ex]: Same in here with escaping

File: lib/ecto/adapters/myxql/connection.ex (hunk 1 above)
Comment (wojtekmach) [reply lib/ecto/adapters/myxql/connection.ex]: Yes, see: https://github.com/elixir-ecto/ecto/pull/3227#discussion_r377126347

**Comment by wojtekmach** [comment]
@michalmuskala You were right, the way to go was properly escaping strings that ended up as JSON keys and not trying to parameterise them. Both adapters had an `escape_string`, I could have adjusted that but decided to introduce a similar `escape_json_key` function that does the same + more. I've updated this (and the other PR) accordingly and added some tests around it in Ecto integration tests. Feedback welcome!

**Comment by josevalim** [comment]
We need some unit tests and then this is ready to go!

**Comment by wojtekmach** [comment]
There was a bug that I realised when writing the unit tests, thank you!

**Pull request merged**